network/peer/ip.go

```diff
@@ -39,7 +39,15 @@ func (ip *UnsignedIP) Sign(tlsSigner crypto.Signer, blsSigner bls.Signer) (*Sign
 		hashing.ComputeHash256(ipBytes),
 		crypto.SHA256,
 	)
-	blsSignature := blsSigner.SignProofOfPossession(ipBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	blsSignature, err := blsSigner.SignProofOfPossession(ipBytes)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Won't let me comment on the line, but we either need to return `nil` at the end of this signature (currently returns `err`), or keep the previous behavior where we don't do the `nil` check and return the `err` alongside `SignedIP`. I personally don't like returning errors with non-nil/zero values, but I think Stephen prefers it because he thinks it looks cleaner/less LOC.